app/src/main/java/com/kms/cura/view/activity/BookAppointmentActivity.java

```diff
@@ -0,0 +1,190 @@
+        button.setOnClickListener(this);
+        return button;
+    }
+
+    public void initView() {
+        btnChooseDate = initButton(R.id.btnChooseDate);
+        btnChooseTime = initButton(R.id.btnChoseTime);
+        btnRequestAppt = initButton(R.id.btnRequestAppt);
+        edtComment = (EditText) findViewById(R.id.edtComment);
+        btnBack = (ImageView) findViewById(R.id.btnBack);
+        btnBack.setOnClickListener(this);
+        initTextView();
+        setUpData();
+        setUpSpinner();
+    }
+
+    public void setUpSpinner() {
+        initSpinner();
+        StringListAdapter adapterFacility = new StringListAdapter(this, R.layout.string_list_item,
+                (ArrayList) doctorFacilityName, facilitySelected);
+        spnFacilities.setAdapter(adapterFacility);
+        spnFacilities.setSelection(adapterFacility.getCount());
+
+        StringListAdapter adapterApptLength = new StringListAdapter(this, R.layout.string_list_item,
+                (ArrayList) apptLength, apptLengthSelected);
+        spnApptLength.setAdapter(adapterApptLength);
+        spnApptLength.setSelection(adapterApptLength.getCount());
+    }
+
+    public void initSpinner() {
+        spnApptLength = (Spinner) findViewById(R.id.spnAppLength);
+        spnFacilities = (Spinner) findViewById(R.id.spnFacility);
+    }
+
+    public void setUpData() {
+        String[] length = getResources().getStringArray(R.array.ApptLength);
+        apptLength = new ArrayList<>();
+        Collections.addAll(apptLength, length);
+        apptLength.add(HINT_TEXT);
+        doctorFacilityName = new ArrayList<>();
+        doctorFacilityName.add("A");
+        doctorFacilityName.add("B");
+        doctorFacilityName.add("C");
+        doctorFacilityName.add(HINT_TEXT);
+    }
+
+    public void initTextView() {
+        txtDate = (TextView) findViewById(R.id.txtDate);
+        txtTime = (TextView) findViewById(R.id.txtTime);
+        edtComment = (EditText) findViewById(R.id.edtComment);
+    }
+
+    @Override
+    public void onClick(View v) {
+        int id = v.getId();
+        if (id == R.id.btnChooseDate) {
+            setCurrentDate();
+            Dialog dateDialog = new DatePickerDialog(this, AlertDialog.THEME_HOLO_LIGHT, myDateListener, year, month - 1, day);
+            dateDialog.show();
+        } else if (id == R.id.btnChoseTime) {
+            if(spnApptLength.getSelectedItemPosition()!= apptLength.size()-1 &&
+                    spnFacilities.getSelectedItemPosition() != doctorFacilityName.size() -1 && selectDate ) {
+                Intent toChooseTime = new Intent(this, ChooseTimeActivity.class);
+                Bundle bundle = createBundle();
+                toChooseTime.putExtras(bundle);
+                startActivityForResult(toChooseTime, REQUEST_cODE);
+            }
+            else {
+                Toast.makeText(this, "Facility, Date and Appointment Length can not be left empty", Toast.LENGTH_SHORT).show();
```

Comment:
"Facility, Date and Appointment Length can not be left empty" should be in resource file